src/node_quic_socket.cc

```diff
@@ -582,9 +582,13 @@ BaseObjectPtr<QuicSession> QuicSocket::AcceptInitialPacket(
   // Check to see if the number of connections for this peer has been exceeded.
   // If the count has been exceeded, shutdown the connection immediately
   // after the initial keys are installed.
-  if (GetCurrentSocketAddressCounter(addr) >= max_connections_per_host_) {
-    Debug(this, "Connection count for address exceeded");
-    initial_connection_close = NGTCP2_SERVER_BUSY;
+  {
+    sockaddr_storage storage;
+    memcpy(&storage, addr, SocketAddress::GetAddressLen(addr));
+    if (GetCurrentSocketAddressCounter(&storage) >= max_connections_per_host_) {
```

Comment:
Try to use `sockaddr_storage` when possible but still have to make a copy here.

Reply:
Do you know why? That seems like it might hide a bug…

Reply:
Because the `addr` is a `sockaddr*` from uv and I think it's not safe to use its value as an `sockaddr_storage` here.